lib/ansible/parsing/ajson.py

```diff
@@ -67,6 +68,8 @@ def default(self, o):
         elif isinstance(o, (date, datetime)):
             # date object
             value = o.isoformat()
+        elif isinstance(o, bytes):
+            value = base64.b64encode(o).decode('ascii')
```

Comment:
I question thing singular change. There is no unique identifier that indicates that base64 encoded data is inherently binary data in JSON.  Additionally, there is no equivalent to this in `AnsibleJSONDecoder`.  As it stands, this is an arbitrary encoding.

I think if we want to properly allow for yaml->to json->from json this should be supported like vault and unsafe. Where we encode it into a dict that properly represents intent, such as making this `__ansible_bytes`

Additionally, this should use `to_text` instead of `.decode`

Reply:
I agree with this. We handled this in `vmware_vm_inventory` since vCenter provides a clear indication of binary data type - https://github.com/ansible-collections/vmware/pull/202